plugins/plugin-java/che-plugin-java-ext-debugger-java-client/src/main/java/org/eclipse/che/ide/ext/java/jdi/client/debug/remotedebug/RemoteDebugPresenter.java

```diff
@@ -39,15 +59,44 @@ public RemoteDebugPresenter(RemoteDebugView view, DebuggerPresenter debuggerPres
     /** Calls special method on view which shows dialog window. */
     public void showDialog() {
         view.show();
+        setPortsList();
+
         // TODO fix behaviour. Because of animation/render we cannot set focus without delay
         new Timer() {
             @Override
             public void run() {
-                view.setFocusInHostField();
+                view.focus();
             }
         }.schedule(300);
     }
 
+    private void setPortsList() {
+        machineServiceClient.getMachine(appContext.getDevMachineId()).then(new Operation<MachineDto>() {
+            @Override
+            public void apply(MachineDto machineDto) throws OperationException {
+                Machine machine = entityFactory.createMachine(machineDto);
+                Map<String, String> ports = extractPortsList(machine);
+                view.setPortsList(ports);
+            }
+        });
+    }
+
+    /**
+     * Extracts list of ports available to connect by debugger.
+     */
+    private Map<String, String> extractPortsList(Machine machine) {
```

Comment:
As for me, it would be better to create simple bean, which will contains two fields, `description` or `host` and `port`.

- At first, it would be more clearly for end user, what exact data is proceed.

- At second, this bean may be used in other third-party components.

WDYT?